### PR TITLE
[Dev Docs] Date picker docs

### PR DESCRIPTION
**Pull request by rami-a**

### CODE REVIEW

File: docs/components/dartpad/date_pickers/range/main.dart [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      lastDate: DateTime(2022, 7),
+      helpText: 'Select a date range',
+    );
+    if (newDate != null) {
+      setState(() {
+        _date = newDate;
+      });
+    }
+  }
+
```
Comment (johnsonmh) [inline docs/components/dartpad/date_pickers/range/main.dart:45]: You should probably add an `@override` before build here.

File: docs/components/dartpad/date_pickers/regular/main.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      lastDate: DateTime(2022, 7),
+      helpText: 'Select a date',
+    );
+    if (newDate != null) {
+      setState(() {
+        _date = newDate;
+      });
+    }
+  }
+
```
Comment (johnsonmh) [inline docs/components/dartpad/date_pickers/regular/main.dart:39]: `@override` here too

File: docs/components/date_pickers.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+![Date pickers anatomy diagram](assets/date_pickers/pickers_anatomy.png)
+
+1. Title
+2. Selected date
+3. Switch-to-keyboard input icon
+4. Year selection menu
+5. Month pagination
+6. Current date
+7. Selected date
```
Comment (JoseAlba) [inline docs/components/date_pickers.md:83]: Nit: In a lot of our docs we don't use 1.2.3.4. we do 1.1.1.1 Markdown is able to read it as a list and easier for us to maintain

File: docs/components/dartpad/date_pickers/theme/main.dart [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      lastDate: DateTime(2022, 7),
+      helpText: 'Select a date',
+    );
+    if (newDate != null) {
+      setState(() {
+        _date = newDate;
+      });
+    }
+  }
+
```
Comment (johnsonmh) [inline docs/components/dartpad/date_pickers/theme/main.dart:39]: `@override` here too

File: docs/components/date_pickers.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+![Date pickers anatomy diagram](assets/date_pickers/range_pickers_anatomy.png)
+
+1. Title
+2. Selected date range
+3. Switch to keyboard input icon
+4. Month and year label
+5. Current date
+6. Start date
+7. Selected range
+8. End date
```
Comment (JoseAlba) [inline docs/components/date_pickers.md:149]: nit: same as previous comment. 1.2.3.4 => 1.1.1.1

**Review by JoseAlba — APPROVED** [review]
```
██╗      ██████╗ ████████╗███╗   ███╗
██║     ██╔════╝ ╚══██╔══╝████╗ ████║
██║     ██║  ███╗   ██║   ██╔████╔██║
██║     ██║   ██║   ██║   ██║╚██╔╝██║
███████╗╚██████╔╝   ██║   ██║ ╚═╝ ██║
╚══════╝ ╚═════╝    ╚═╝   ╚═╝     ╚═╝
```

File: docs/components/date_pickers.md [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+      lastDate: DateTime(2022, 7),
+      helpText: 'Select a date',
+    );
+    if (newDate != null) {
+      setState(() {
+        _date = newDate;
+      });
+    }
+  }
+
```
Comment (johnsonmh) [inline docs/components/date_pickers.md:217]: nit: @override here

**Review by johnsonmh — APPROVED** [review]
LGTM

File: docs/components/date_pickers.md [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+Mobile date pickers allow selection of a single date and year.
+
+### Date pickers example
+
+`showDatePicker`
+- [Class definition](https://api.flutter.dev/flutter/material/showDatePicker.html)
+- [GitHub souce](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/pickers/date_picker_dialog.dart)
+- [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/date_pickers/regular&gh_ref=develop)
+
+The following example shows a date picker.
```
Comment (mingjane-work) [inline docs/components/date_pickers.md:57]: Please update to:

"The following example shows a date picker with a selected date of November 17, 2020 and a current date of November 13, 2020."

File: docs/components/date_pickers.md [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+
+![Date picker with date set to November 17, 2020](assets/date_pickers/date_picker_example.png)
```
Comment (mingjane-work) [inline docs/components/date_pickers.md:59]: Please update alt-text to:

"Date picker with selected date of November 17, 2020 and current date of November 13, 2020"

File: docs/components/date_pickers.md [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+### Date range picker example
+
+`showDateRangePicker`
+- [Class definition](https://api.flutter.dev/flutter/material/showDateRangePicker.html)
+- [GitHub souce](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/pickers/date_range_picker_dialog.dart)
+- [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/date_pickers/range&gh_ref=develop)
+
+The following example shows a date range picker.
+
+![Date range picker with dates set from November 17, 2020 to November 24, 2020](assets/date_pickers/date_range_picker_example.png)
```
Comment (mingjane-work) [inline docs/components/date_pickers.md:121]: Please update alt-text to:

"Date range picker selecting November 17, 2020 to November 24, 2020, current date of November 13, 2020."

**Pull request merged**